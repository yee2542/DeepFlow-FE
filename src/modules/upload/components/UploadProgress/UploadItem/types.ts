import { GroupTask } from 'Modules/upload/model/types';

type UploadItemProps = {
  file: {
    file: GroupTask;
  };
};

type ProgressBarPropsType = {
  width: number;
};

export type { UploadItemProps, ProgressBarPropsType };
