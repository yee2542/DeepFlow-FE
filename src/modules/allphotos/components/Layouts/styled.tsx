import styled from 'styled-components';
import { Layout } from 'antd';
const { Sider } = Layout;

export const CanvasWrapper = styled.div`
  .ant-layout-header {
    background-color: white;
  }

  .ant-layout-content {
    height: 100vh;
    background-color: white;
  }
`;

export const SiderWrapper = styled(Sider)`
  background-color: #4425a7;
  width: 294px;
  padding: 2%;
`;

export const MenuWrapper = styled.div`
  margin-top: 20%;
  width: 100%;
`;

export const Menu = styled.div`
  margin-top: 20%;
  a:hover {
    padding: 10%;
    border-radius: 2em;
    background-color: #48cdd8;
  }
  a:active {
    padding: 10%;
    border-radius: 2em;
    background-color: #48cdd8;
    position: inherit;
  }
  a {
    color: white;
  }
  width: 100%;
`;
