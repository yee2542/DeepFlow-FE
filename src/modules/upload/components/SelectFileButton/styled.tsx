import styled from 'styled-components';

export const UploadBtnWrapper = styled.div`
  input[type='file'] {
    width: 0.1px;
    height: 0.1px;
    opacity: 0;
    overflow: hidden;
    position: absolute;
    z-index: -1;
  }

  input[type='file'] + label {
    font-size: 1.25em;
    font-weight: 700;
    color: #5a36cc;
    background-color: #f8f5ff;
    border: 1px solid #d5c8ff;
    display: inline-block;
    cursor: pointer;
    border-radius: 4px;
    padding: 30px 40px 30px 40px;
    text-align: center;
  }

  input[type='file'] + label:hover {
    background-color: #ebe4ff;
  }
`;
