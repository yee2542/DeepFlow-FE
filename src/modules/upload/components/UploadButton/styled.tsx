import styled from 'styled-components';

export const LabelUpload = styled.label`
  background-color: var(--secondary-color);
  color: white;
  padding: 1rem;
  font-family: sans-serif;
  border-radius: 1.5rem;
  cursor: pointer;
  margin-top: 1rem;
`;
