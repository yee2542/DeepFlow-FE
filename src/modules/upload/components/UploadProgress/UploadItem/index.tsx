import React from 'react';
import { WrapperItem, LeftSide, ProgressBar, Percentage } from './styled';
import { UploadItemProps } from './types';

const UploadItem: React.FC<UploadItemProps> = (props) => {
  const { file } = props.file;
  return file ? (
    <WrapperItem>
      <LeftSide>
        <ProgressBar>
          <div style={{ width: `${file.progress}%` }} />
          <label>{file.label}</label>
        </ProgressBar>
      </LeftSide>
      <Percentage>{file.progress}</Percentage>
    </WrapperItem>
  ) : null;
};

export default UploadItem;
