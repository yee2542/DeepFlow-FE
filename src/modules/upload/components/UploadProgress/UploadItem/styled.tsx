import styled from 'styled-components';
import { ProgressBarPropsType } from './types';

export const WrapperItem = styled.div`
  display: grid;
  grid-template-columns: calc(100% - 50px) 50px;
  padding: 16px;
  box-shadow: 0px 1px 2px lightgray;
`;

export const LeftSide = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-right: 8px;

  label {
    width: 100%;
    overflow: hidden;
    text-overflow: ellipsis;
    text-align: left;
  }
`;

export const ProgressBarWrapper = styled.div`
  width: 100%;
  height: 15px;
  background-color: lightgray;
  margin-bottom: 8px;
  border-radius: 20px;

  div {
    height: 15px;
    background-color: lightgreen;
    border-radius: 20px;
  }
`;

export const Percentage = styled.span`
  margin-left: 12px;
`;

export const ProgressBar = styled.div<ProgressBarPropsType>`
  ${(props) =>
    props.width &&
    `
   width: ${props.width}%`}
`;
