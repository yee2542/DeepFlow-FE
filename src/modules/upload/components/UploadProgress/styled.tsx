import styled from 'styled-components';

export const Wrapper = styled.div`
  position: fixed;
  bottom: 0;
  right: 0;
  background-color: white;
  width: 400px;
  overflow: scroll;
  max-height: 400px;
  border-top-left-radius: 8px;
  border-top-right-radius: 8px;
  h5 {
    background-color: var(--secondary-color);
    margin: 0;
    color: white;
    padding: 20px;
  }
`;
