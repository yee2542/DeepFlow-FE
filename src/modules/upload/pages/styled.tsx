import styled from 'styled-components';
import { Layout } from 'antd';

const { Footer } = Layout;

export const BottomBar = styled(Footer)`
  background-color: white;
  box-shadow: rgba(100, 100, 111, 0.3) 0px 7px 29px 0px;
  position: fixed;
  height: 6rem;
  bottom: 0;
  width: 100%;
  text-align: right;
`;

export const UploadWrapper = styled.div`
  height: 100vh;
  padding: 5%;
  background-color: white;

  .ant-layout-content {
    background-color: white;
  }
`;

export const Image = styled.img`
  width: 100%;
  height: 100%;
  object-fit: cover;
`;

export const BackButton = styled.button`
  background: none;
  color: var(--text-color-secondary);
  border: none;
  padding: 1rem;
  font: inherit;
  cursor: pointer;
  outline: inherit;
  border-radius: 50%;

  &:hover {
    background-color: #ededed;
    padding: 1rem;
    border-radius: 50%;
  }
`;
