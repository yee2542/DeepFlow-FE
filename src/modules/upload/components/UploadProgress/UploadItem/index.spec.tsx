import React from 'react';
import UploadItem from '.';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import { ProgressBar } from './styled';

describe('Component/UploadItem test', () => {
  const mockFile = { file: { label: 'hello', progress: 10, total: 100 } };
  it('Should match snapshots', () => {
    const wrap = renderer.create(<UploadItem file={mockFile} />).toJSON();
    expect(wrap).toMatchSnapshot();
  });

  it('UploadItem should have file props with file', () => {
    const wrap = mount(<UploadItem file={mockFile} />);
    expect(wrap.props().file).toBe(mockFile);
  });

  it('UploadItem progress should be changed equals to input file progress', () => {
    const wrap = shallow(<UploadItem file={mockFile} />);
    const progressBar = wrap.find(ProgressBar);
    expect(progressBar.props().width).toBe(mockFile.file.progress);
  });
});
