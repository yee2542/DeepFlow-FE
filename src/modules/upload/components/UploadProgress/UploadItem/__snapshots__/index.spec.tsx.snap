// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/UploadItem test Should match snapshots 1`] = `
<div
  className="styled__WrapperItem-ukzgv1-0 efQqci"
>
  <div
    className="styled__LeftSide-ukzgv1-1 jWWiFz"
  >
    <div
      className="styled__ProgressBarWrapper-ukzgv1-2 gPOYAe"
    >
      <div
        className="styled__ProgressBar-ukzgv1-4 gvwXJY"
        width={10}
      />
      <label>
        hello
      </label>
    </div>
  </div>
  <span
    className="styled__Percentage-ukzgv1-3 lojdNb"
  >
    10
  </span>
</div>
`;
